### PR TITLE
Create a baggage reference implementation

### PR DESCRIPTION
**Pull request by dyladan**
- [x] reference implementation
- [x] unit test reference implementation
- [x] unit test limits

Create a client-server test harness in a followup PR which will allow arbitrary implementations to be tested

### CODE REVIEW

File: test/baggage/test_baggage.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import unittest
+import urllib.parse
+
+from baggage import Baggage, BaggageEntry
+
+
+class BaggageTest(unittest.TestCase):
```
Comment (kalyanaj) [inline test/baggage/test_baggage.py:7]: This is a good set of tests, thanks for adding them. Are you planning to add any limits related tests, or is the plan to add them after we finish the discussion around min/max limits?

**Review by kalyanaj — APPROVED** [review]

File: test/baggage/test_baggage.py (hunk 1 above)
Comment (dyladan) [reply test/baggage/test_baggage.py:7]: Yes the plan is to add limits, and to add a client-server test harness

**Comment by dyladan** [comment]
Hoping for a couple more reviews of this and maybe get the above question answered so I know if I have to do more work on this or not.

**Comment by dyladan** [comment]
Still need more reviews on this

**Review by basti1302 — APPROVED** [review]
LGTM

**Pull request merged**